.github/workflows/common.yml

```diff
@@ -0,0 +1,115 @@
+---
+# Source: https://github.com/pharmaverse/admiralci
+# Common workflows designed for Admiral
+# but can be easily used by any other R package
+name: Admiral Workflows
+
+on:
+  # 'workflow_dispatch' gives you the ability
+  # to run this workflow on demand, anytime
+  workflow_dispatch:
+  # 'push' events are triggered when commits
+  # are pushed to one of these branches
+  push:
+    branches:
+      - main
+      - devel
+      - pre-release
+      - test
+  # 'pull_request' events are triggered when PRs are
+  # created against one of these target branches.
+  pull_request:
+    branches:
+      - main
+      - devel
+      - pre-release
+      - test
+  # 'release' events are triggered when...
+  # you guessed it - when releases are made.
+  release:
+    types: [published]
+
+env:
+  # R version to use for the workflows
+  R_VERSION: "3.6"
+
+# Docs on concurrency:
+# https://docs.github.com/en/actions/using-jobs/using-concurrency
+concurrency:
+  group: admiral-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  style:
+    name: Code Style
+    uses: pharmaverse/admiralci/.github/workflows/style.yml@main
+    if: github.event_name == 'pull_request'
+    with:
+      r-version: $R_VERSION
+  spellcheck:
+    name: Spelling
+    uses: pharmaverse/admiralci/.github/workflows/spellcheck.yml@main
+    if: github.event_name == 'pull_request'
+    with:
+      r-version: $R_VERSION
+  readme:
+    name: Render README
+    uses: pharmaverse/admiralci/.github/workflows/readme-render.yml@main
+    if: github.event_name == 'push'
+    with:
+      r-version: $R_VERSION
+  validation:
+    name: Validation
+    uses: pharmaverse/admiralci/.github/workflows/r-pkg-validation.yml@main
+    if: github.event_name == 'release'
+    with:
+      r-version: $R_VERSION
+  check:
+    name: Check
+    uses: pharmaverse/admiralci/.github/workflows/r-cmd-check.yml@main
+    if: github.event_name == 'pull_request'
```

Comment:
I assume that still runs on multiple R versions? What about the package versions? In our current `R CMD check` workflow we use MRAN snapshots from the date of the release of the tested R version. I'd very much like to continue doing that going forward.